nwcsafpps_runner/pps_posttroll_hook.py

```diff
@@ -112,19 +116,21 @@ class PPSMessage(object):
 
     """
 
-    def __init__(self, station, posttroll_topic, output_format, level):
+    # def __init__(self, software, geo_or_polar, variant, output_format, level):
+    def __init__(self, description, metadata):
 
         # __init__ is not run when created from yaml
         # See http://pyyaml.org/ticket/48
         pass
 
     def __getstate__(self):
         """Example - metadata:
-        posttroll_topic: "PPSv2018"
         station: "norrkoping"
         output_format: "CF"
         level: "2"
         variant: DR
+        geo_or_polar: "polar"
+        software: "NWCSAF-PPSv2018"
         """
```

Comment:
I have not understood the `__getstate__` function. But is it really necessary to hardcode geo_or_polar nad software?

Reply:
I am not sure I remember anymore, but the metadata here is used in the publish messages and I think we needed to separate SEVIRI from the polar production. We will need to iterate what is needed as mandatory parameters, once you start using this. I have tries thinking carefully what is needed and what is not. But it might need to change....